httpcore5/src/main/java/org/apache/hc/core5/pool/StrictConnPool.java

```diff
@@ -171,21 +171,36 @@ public void close() {
         Args.notNull(route, "Route");
         Args.notNull(requestTimeout, "Request timeout");
         Asserts.check(!this.isShutDown.get(), "Connection pool shut down");
+        final Deadline deadline = Deadline.calculate(requestTimeout);
         final BasicFuture<PoolEntry<T, C>> future = new BasicFuture<>(callback);
-        this.lock.lock();
+        final boolean acquiredLock;
+
         try {
-            final LeaseRequest<T, C> request = new LeaseRequest<>(route, state, requestTimeout, future);
-            final boolean completed = processPendingRequest(request);
-            if (!request.isDone() && !completed) {
-                this.leasingRequests.add(request);
-            }
-            if (request.isDone()) {
-                this.completedRequests.add(request);
+            acquiredLock = this.lock.tryLock(requestTimeout.getDuration(), requestTimeout.getTimeUnit());
+        } catch (final InterruptedException interruptedException) {
+            Thread.currentThread().interrupt();
+            future.failed(interruptedException);
```

Comment:
@cwildman One last minor thing. Would not `future#cancel()` be more appropriate here?

Reply:
Yeah that does seem right.